tools/project_manager/editproject.php

```diff
@@ -688,18 +691,20 @@ function save_to_db()
             }
 
             // Update the projects database with the updated info
-            mysqli_query(DPDatabase::get_connection(), "
+            $sql = sprintf("
                 UPDATE projects SET
                     $pm_setter
                     $tlcc_setter
                     $md_setter
                     $common_project_settings
```

Comment:
There is no guarantee that the contents of `$common_project_settings` doesn't contain a `%`, so we can't use `sprintf()` here like this. I really hate how we build queries up like this in so many places.

You could do:
```php
            $where = sprintf("projectid='%s'", DPDatabase::escape($this->projectid));
            $sql = "
                UPDATE projects SET
                    $pm_setter
                    $tlcc_setter
                    $md_setter
                    $common_project_settings
                WHERE $where";
```

Reply:
Oh that's hideous. Yeah, I'll keep that in mind in future.